ui/src/app/shared/components/page.tsx

```diff
@@ -18,7 +18,7 @@ export class Page extends React.Component<{ title: string, toolbar?: Toolbar | O
                 toolbar = toolbar || {};
                 toolbar.tools = [
                     toolbar.tools,
-                    services.authService.getCurrentUserId() ?
+                    parse(document.cookie)['argocd.token'] ?
```

Comment:
kind of a crummy check, but when I made this bit of code async then the UI goes loco and flashing

Reply:
ok :(